CriteoPublisherSdk/Tests/IntegrationTests/CR_ProfileIdFunctionalTests.m

```diff
@@ -312,4 +327,17 @@ - (NSDictionary *)csmRequest {
   }];
 }
 
+- (void)enrichAd:(id)adObject
+       forAdUnit:(CRAdUnit *)adUnit
+     thenWaitFor:(void (^)(void))completionBlock {
+  [self cr_waitForExpectationBlock:^(XCTestExpectation *expectation) {
```

Comment:
Same comment than in CR_IntegrationTestBase. What do you think about mocking a synchronous call to make test easier to read ?

I see that test are built with `waitForIdleState`. Then tests should already be sync, no ?

Reply:
Same as previous comment, not all tests are based on idle state, this approach works for all scenarios.

Reply:
Done